src/ucs/sys/sock.c

```diff
@@ -100,22 +100,74 @@ ucs_status_t ucs_netif_ioctl(const char *if_name, unsigned long request,
     return status;
 }
 
-int ucs_netif_is_active(const char *if_name)
+ucs_status_t ucs_netif_get_addr(const char *if_name, uint8_t sa_family,
+                                struct sockaddr *saddr,
+                                struct sockaddr *netmask)
 {
-    ucs_status_t status;
-    struct ifreq ifr;
+    ucs_status_t status = UCS_ERR_NO_DEVICE;
+    struct ifaddrs *ifa;
+    struct ifaddrs *ifaddrs;
+    const struct sockaddr_in6 *saddr6;
+    size_t addrlen;
 
-    status = ucs_netif_ioctl(if_name, SIOCGIFADDR, &ifr);
-    if (status != UCS_OK) {
-        return 0;
+    if (getifaddrs(&ifaddrs)) {
+        ucs_warn("getifaddrs error: %m");
+        status = UCS_ERR_IO_ERROR;
+        goto out;
     }
 
-    status = ucs_netif_ioctl(if_name, SIOCGIFFLAGS, &ifr);
-    if (status != UCS_OK) {
-        return 0;
+    for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+        if ((if_name != NULL) && (0 != strcmp(if_name, ifa->ifa_name))) {
+            continue;
+        }
+
+        if ((ifa->ifa_addr == NULL) ||
+            ((ifa->ifa_addr->sa_family != AF_INET) &&
+             (ifa->ifa_addr->sa_family != AF_INET6))) {
+            continue;
+        }
+
+        if (!ucs_netif_flags_is_active(ifa->ifa_flags)) {
+            continue;
+        }
+
+        if (ifa->ifa_addr->sa_family == AF_INET6) {
+            saddr6 = (const struct sockaddr_in6*)ifa->ifa_addr;
+            if (IN6_IS_ADDR_LOOPBACK(&saddr6->sin6_addr) ||
+                IN6_IS_ADDR_LINKLOCAL(&saddr6->sin6_addr)) {
```

Comment:
shall we check the same for IPv4 too?
if yes better to use `ucs_sockaddr_is_inaddr_loopback()` which is common for IPv4 and IPv6.

Reply:
it was not ignoring loopback for ipv4 earlier. didn't want change that behavior